argilla/docs/how_to_guides/webhooks.md

```diff
@@ -0,0 +1,162 @@
+---
+description: In this section, we will provide a step-by-step guide to create a webhook in Argilla.
+---
+
+# Webhooks
+
+In this section, we will provide a step-by-step guide to create a webhook in Argilla.
+
+## What is a webhook?
+
+A webhook is a way for an app to provide other applications with real-time information. A webhook delivers data to other applications as it happens, meaning you get data immediately. Unlike typical APIs where you would need to poll for data very frequently in order to get it real-time.
+This makes webhooks much more efficient for both provider and consumer.
+
+## How to create a webhook in Argilla?
+
+The python SDK provides a simple way to create a webhook in Argilla. Its allow you focus on the use case of the webhook
+and not on the implementation details. You only need to create your event handler function with the `webhook_listener` decorator.
+
+```
+import argilla as rg
+
+from datetime import datetime
+from argilla_sdk import webhook_listener
+
+@webhook_listener(events="dataset.created")
+async def my_webhook_handler(dataset: rg.Dataset, type: str, timestamp: datetime):
+    print(event)
+```
+
+In the example above, we have created a webhook that listens to the `dataset.created` event. (You can find the list of events in the [Events](#events) section).
+
+The python SDK will automatically create a webhook in Argilla and listen to the specified event. When the event is triggered,
+the `my_webhook_handler` function will be called with the event data. The SDK will also parse the incoming webhook event into
+a proper resource object (rg.Dataset, rg.Record, and rg.UserResponse). The SDK will also take care of request authentication and error handling.
```

Comment:
```suggestion
a proper resource object (`rg.Dataset`, `rg.Record`, and `rg.Response`). The SDK will also take care of request authentication and error handling.
```
Users don't normally use UserResponse directly right?

Reply:
Yes, I think this might confuse user. Seeing a random new resource that they normally don't deal with. Either way, let's leave it for now, and we'll see if they actually do.